plugin.js

```diff
@@ -0,0 +1,139 @@
+const path = require("path");
+const fs = require("fs");
+const shell = require("shelljs");
+const { JSDOM } = require("jsdom");
+
+const Config = require("./Config.js");
+const { PluginScope } = require("./Enums.js");
+
+function setCurr(component, targetPath, subTaskFlag=false) {
+	let obj = {...component}, isCurrentItem = false;
+
+	if(obj.unit_type === "aim")
+	{
+		subTaskFlag = true;
+		isCurrentItem = true;
+	}
+
+	if(!subTaskFlag)
+	{
+		obj = component.menuItemInfo(targetPath);
+	}
+
+	if(obj.unit_type === "lu")
+	{
+		obj.units = [...obj.units.map((subComponent) => setCurr(subComponent, targetPath, true))];
+	}
+
+	return { ...obj, isCurrentItem: isCurrentItem };
+};
+
+class Plugin {
+
+  static getConfigFileName(options_env) {
+    const env = options_env || BuildEnvs.TESTING;
+    const pluginConfigFile = `./plugin-config.${env}.js`;
+    return pluginConfigFile;
+  }
+
+  static processExpScopePlugins(exp_info, hb, lab_data, options) {
+    const env = options.env || BuildEnvs.TESTING;
+    const pluginConfigFile = `./plugin-config.${env}.js`;
+    const pluginConfig = require(pluginConfigFile);
+
+    const expScopePlugins = pluginConfig.filter(
+      (p) => p.scope === PluginScope.EXPERIMENT
+    );
+
+    let plugins = [];
+    shell.exec('mkdir plugins');
+    expScopePlugins.forEach((plugin) => {
+	    try {
+		    if(!fs.existsSync(path.join('plugins', plugin.id)))
+		    {
+			    shell.cd('plugins');
+			    shell.exec('git clone ' + plugin.repo);
+			    shell.exec(`rm -rf ${path.join(plugin.id, '.git')}`);
+			    shell.cd('..');
+		    }
+		    shell.exec('cp -ur \'' + path.resolve('./plugins') + '\' \'' + Config.build_path(exp_info.src) + '\'');
```

Comment:
I am not sure what this command is doing. We are recursive copying the plugins directory over to the src directory. Why don't we create the directory there in the first place. Will be cleaner and easier to manage with clean command etc.